cluster-up/cluster/kind-k8s-sriov-1.17.0/deploy_sriov_operator.sh

```diff
@@ -0,0 +1,254 @@
+    export SRIOV_CNI_IMAGE=quay.io/openshift/origin-sriov-cni:${RELEASE_VERSION}
+    export SRIOV_DEVICE_PLUGIN_IMAGE=quay.io/openshift/origin-sriov-network-device-plugin:${RELEASE_VERSION}
+    export OPERATOR_EXEC=${KUBECTL}
+    make deploy-setup-k8s SHELL=/bin/bash  # on prow nodes the default shell is dash and some commands are not working
+  popd
+
+  echo 'Generating webhook certificates for the SR-IOV operator webhooks'
+  pushd "${CERTCREATOR_PATH}"
+    go run . -namespace sriov-network-operator -secret operator-webhook-service -hook operator-webhook -kubeconfig $KUBECONFIG_PATH
+    go run . -namespace sriov-network-operator -secret network-resources-injector-secret -hook network-resources-injector -kubeconfig $KUBECONFIG_PATH
+  popd
+
+  echo 'Setting caBundle for SR-IOV webhooks'
+  wait_k8s_object "validatingwebhookconfiguration" "operator-webhook-config"
+  _kubectl patch validatingwebhookconfiguration operator-webhook-config --patch '{"webhooks":[{"name":"operator-webhook.sriovnetwork.openshift.io", "clientConfig": { "caBundle": "'"$(cat $CERTCREATOR_PATH/operator-webhook.cert)"'" }}]}'
+
+  wait_k8s_object "mutatingwebhookconfiguration"   "operator-webhook-config"
+  _kubectl patch mutatingwebhookconfiguration operator-webhook-config --patch '{"webhooks":[{"name":"operator-webhook.sriovnetwork.openshift.io", "clientConfig": { "caBundle": "'"$(cat $CERTCREATOR_PATH/operator-webhook.cert)"'" }}]}'
+
+  wait_k8s_object "mutatingwebhookconfiguration"   "network-resources-injector-config"
+  _kubectl patch mutatingwebhookconfiguration network-resources-injector-config --patch '{"webhooks":[{"name":"network-resources-injector-config.k8s.io", "clientConfig": { "caBundle": "'"$(cat $CERTCREATOR_PATH/network-resources-injector.cert)"'" }}]}'
+
+  return 0
+}
+
+function apply_sriov_node_policy {
+  local -r policy_file=$1
+  local -r pfs_names_json_arrays=$2
+  local -r num_vfs=$3
+  
+  # Format to singal array ["ens1","ens2"],["ens3"] => ["ens1","ens2","ens3"]
+  local all_pfs_names_json_array=$(sed 's/\].*\[/"," /g' <<< "$pfs_names_json_arrays")
+  
+  # Patch SriovNetworkNodesPolicy with PF's names and VF's count
+  sed -i "s?pfNames:.*?pfNames: $all_pfs_names_json_array?g" $policy_file
+  sed -i "s?numVfs:.*?numVfs: $num_vfs?g" $policy_file
+
+  echo "Create SriovNetworkNodeConfigPolicy"
+  cat $policy_file
+  kubectl create -f $policy_file
+
+  return 0
+}
+
+deploy_multus
+wait_pods_ready
+
+deploy_sriov_operator
+wait_pods_ready
+
+# Patch SriovNetworkNodesPolicy
+policy_template="$MANIFESTS_DIR/network_config_policy.yaml"
+policy="${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/network_config_policy.yaml"
+cp -f $policy_template $policy
+
+used_pfs_names_annotations=$(_kubectl get nodes -l $SRIOV_NODE_LABEL \
+  -o jsonpath="{.items[*].metadata.annotations.$NODE_PFS_ANNOTATION}")
+vfs_count=$(cat /sys/bus/pci/devices/*/sriov_totalvfs | head -1)
```

Comment:
This might ruin the multi sriov, please verify which PFs you see here,
in case you see them all, and not just the ones the undercloud CNI allocated to this job, 
then please dont use /sys/bus/pci, but only the ones in the cluster namespace

Reply:
`sriov_totalvfs` file represent the VF count that configured on the card firmware which is identical to all PF's on the host
thus It should not break multi cluster functionality.

Reply:
You can configure it per PF.
In case we leave it like this its an assumption that at least need to be asserted please.

Reply:
You can configure it per PF.
In case we leave it like this its an assumption, and we better not have it, as there is a simpler solution:
just see how we take the VF count already, by looking in the node itself, this way it will be more robust.

Reply:
You should be able to access the device from the interface: `/sys/class/net/ens1f0/device/sriov_totalvfs`.

You will need to build the policy from a template one pf at a time. But I am ok with asserting that the vf-count is indeed identical on all used pf/s from the list (but not `*`). If it is not, we should fail.

Reply:
`sriov_*total*vfs` is read only file and should not be changed by the user, 
it should be consistent across PF's as it represents the hardware limitation 
it is mentioned at intel's documentation https://www.intel.com/content/www/us/en/embedded/products/networking/sr-iov-guide-ethernet-x710-xl710-linux-brief.html

Reply:
At this point the PF's moved to the nodes netns so it should be invoked from the container
I trying to avoid this, so this script could do only the deployment w/o investigation the nodes.

I dont think we need to create a policy for each PF if this what you mean,
we should create single policy that will include all PF's and the VF's count (which is identical to all PF's)
so we could have single VF's pool for all the cluster at we expect on tests.

Reply:
> it is mentioned at intel's documentation https://www.intel.com/content/www/us/en/embedded/products/networking/sr-iov-guide-ethernet-x710-xl710-linux-brief.html

On Mellanox you can config it per PF.
and even if it wasn't the case, one PF might be corrupted on the machine, and then we wouldn't able to use it, we would black list it, but it might be the first, and mislead us in case its MAX_VF is zero.
I would suggest to make it robust and env agnostic please.

> At this point the PF's moved to the nodes netns so it should be invoked from the container
I trying to avoid this, so this script could do only the deployment w/o investigation the nodes.

I find inspecting the nodes and keeping the code more robust and simpler, better than assuming and using the annotation,
the data is already there, no need to have it twice.
Moreover if the PF is missing for any reason after it was allocated (it might happen due to bugs, especially if a container is deleted), using the annotation is less safe than inspecting the container.